### ISSUE TITLE
place() collapses table/grid inside bulleted list

### ISSUE THREAD

**Issue by raphCode** [typst/typst#2948]
### Description

`place`ing a `grid` or a `table` inside a bulleted list seems to collapse the grid/table when the actual text of the list item is too short.
Repro:
```typst
#let t = table(columns: 3, ..range(9).map(str))

#place(t)
#v(2cm)
- typst #place(t)
#v(2cm)
- #place(t)
```
![image](https://github.com/typst/typst/assets/15750438/5b78d898-6b92-431a-be7b-269489308b98)

### Operating system

Web app

### Typst version

- [X] I am using the latest version of Typst

**Comment by PgBiel** [comment]
As you didn't specify the column sizes (you specified `columns: 3` which is equivalent to `columns: (auto, auto, auto)`), the column sizes will be (approximately) equivalent to the largest width of a cell in each column, but _reduced_ to fit the page (or container) width when needed. Therefore, it appears that, with `#place`, the available width isn't made very clear, so you can force a particular container width by yourself to tell the table it has enough space to expand as much as it needs.

```typ
#let t = table(columns: 3, ..range(9).map(str))

#place(t)
#v(2cm)
- typst #place(block(width: 200pt, t))  // indicate the table has 200pt available to expand (it won't fill it entirely though)
#v(2cm)
- #place(block(width: 200pt, t))
```
![now the tables render correctly](https://github.com/typst/typst/assets/9021226/ae7f397e-c530-475c-8e41-2b0902879d93)

**Comment by raphCode** [comment]
> with #place, the available width isn't made very clear

Ah yes, you are right. (I really should not try to report bugs after midnight and go to sleep instead.)

Seems to happen for all containers whenever the parent container size is automatically determined.
Even something like this gives a zero-width rectangle:
```
#set page(width: auto)
#rect(height: 1cm, width: 100%, stroke: 1pt)
```

**Comment by raphCode** [comment]
Should we close this?

**Comment by laurmaedje** [comment]
I think it's still quirky, so maybe we should leave it open.

**Comment by PgBiel** [comment]
> I think it's still quirky, so maybe we should leave it open.

I have to admit that, indeed, it could be considered a bit odd for `place` to behave like this by default... perhaps it should give more available space to whatever is inside it by default (or maybe even unlimited available space), similar to `measure`?

**Comment by laurmaedje** [comment]
Unlimited would probably be undesirable. E.g. when placing something at the top of the page, you still want it to respect the page width. But something should probably be changed here.

**Comment by PgBiel** [comment]
Right... after investigation, it appears that this is the same problem as https://github.com/typst/typst/issues/529. Consider:

```typ
#let t = table(columns: 3, ..range(9).map(str))

#place(t)
#v(2cm)
- typst #place(t) #h(1fr)
#v(2cm)
- #place(t) #h(1fr)
```

![also renders correctly](https://github.com/typst/typst/assets/9021226/145811a6-68af-49ee-853b-5d2feab7882c)

The list items just aren't expanding to full width unless explicitly prompted to do so, thus not giving enough space for the table, which didn't explicitly request any particular amount of space (because `place` makes it request 0 space).

**Comment by raphCode** [comment]
I think a related problem is that it is impossible to center-align and `place` content at the same time, because place communicates 0 space.
```typst
#place(align(center, {
  //h(1fr)
  rect(fill:blue, width: 1cm, height: 1cm)
  [hi]
  //h(1fr)
})
```

The `h(1fr)` seem to do interesting things here, centering either the whole thing or just the rectangle. Make sure to try all 4 combinations of (un)commenting them.

**Comment by laurmaedje** [comment]
The rect is block-level here, so the whole setup is a bit strange because h is inline.
